install/upgrade.php

```diff
@@ -204,6 +204,65 @@ protected function upgrade_2_7() {
 	 * @return void
 	 */
 	protected function upgrade_3_4() {
+		global $wpdb;
+
 		delete_transient( 'pll_languages_list' );
+
+		if ( ! defined( 'POLYLANG_PRO' ) || ! POLYLANG_PRO ) {
+			return;
+		}
+
+		// Migrate locale fallbacks from term metas to language term description.
+		$metas = $wpdb->get_results(
+			"
+			SELECT m.meta_id, m.term_id, m.meta_value, tt.description
+			FROM {$wpdb->termmeta} AS m
+			LEFT JOIN {$wpdb->term_taxonomy} AS tt
+				ON tt.term_id = m.term_id
+			WHERE
+				tt.taxonomy = 'language'
+				AND m.meta_key = 'fallback'
+			"
+		);
```

Comment:
WHen possible, I prefer to avoid direct SQL queries. This is interesting for critical performance cases but is less readable. Even with 20 languages, I don't thinks the spee would be critical here.

What do you think of:
```php
$args = array(
	'taxonomy' => 'language',
	'hide_empty' => 'false',
);
	
$terms = get_terms( $args );

foreach ( $terms as $term ) {
	$meta = get_term_meta( $term->term_id, 'fallback', true );

	if ( empty( $meta ) ) {
		continue;
	}

	$description = maybe_unserialize( $term->description );
	$description['fallbacks'] = $meta; 
	wp_update_term( (int) $meta->term_id, 'language', array( 'description' => maybe_serialize( $description ) ) );

	delete_term_meta( $term->term_id, 'fallback' );
}
```

Reply:
Done in [7eb63dd](https://github.com/polylang/polylang/pull/1192/commits/7eb63dddb12251bec3b3174a05286563bf774f77).
On second thought, `get_term_meta()` shouldn't add queries since the values are cached by `get_terms()` (arg `update_term_meta_cache`).
It still adds a lot of queries (29 for 3 languages, only 1 with fallbacks) but I don't know why, maybe something else happened at the same time.

Reply:
Ok, the bunch of queries comes from the things hooked to `wp_update_term()` and `delete_term_meta()`. They don't seem to slow things down by a lot though.

Reply:
For sure I knew that your initial proposal was more performant.